text_generation/causal_lm/cpp/continuous_batching/apps/throughput_benchmark.cpp

```diff
@@ -119,6 +124,58 @@ Dataset filtered_dataset(const std::string& models_path, const std::string& data
     return sampled_dataset;
 }
 
+void trafficSimulator(ContinuousBatchingPipeline* pipe, Dataset* dataset, std::string request_rate) {
+    double numeric_request_rate;
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::exponential_distribution<> distribution;
+
+    if (request_rate == "inf") {
+        numeric_request_rate = -1.0;
+    } else {
+        numeric_request_rate = std::stod(request_rate);
+        if (numeric_request_rate < 0)
+            throw std::invalid_argument("request_rate cannot be a negative number");
+
+        distribution = std::exponential_distribution<>(numeric_request_rate);
+    }
+
+    std::cout << "Launching traffic simulator thread with request_rate: " << request_rate << std::endl;
+
+    for (size_t request_id = 0; request_id < dataset->size(); ++request_id) {
+        std::cout << "Traffic thread adding request to the queue..." << std::endl;
+        pipe->add_request(request_id, dataset->m_prompts[request_id], dataset->m_sampling_params[request_id]);
```

Comment:
looks like we need to insert std::mutex internally to protect access to requests queue, because currently 2 threads have an access to it:
- traffic one
- scheduler from LLM engine loop thread